proxy/envoy/resources.go

```diff
@@ -18,96 +18,67 @@ import (
-	ProxyPort int
-
-	// AdminPort is the administrative interface port
-	AdminPort int
-
-	// Envoy binary path
-	BinaryPath string
-
-	// Envoy config root path
-	ConfigPath string
-
-	// Whether Envoy enforces auth for proxy-proxy traffic
-	EnableAuth bool
-
-	// The path for auth config files
-	AuthConfigPath string
-
-	// DrainTimeSeconds is the duration of the grace period to drain connections
-	// from an older proxy instance
-	DrainTimeSeconds int
-
-	// ParentShutdownTimeSeconds is the duration to wait to shutdown the older
-	// proxy instance
-	ParentShutdownTimeSeconds int
+const (
+	// DefaultAccessLog is the name of the log channel (stdout in docker environment)
+	DefaultAccessLog = "/dev/stdout"
 
-	// IstioServiceCluster defines the name for the service_cluster that is
-	// shared by all proxy instances. Since Istio does not assign a local
-	// service/service version to each proxy instance, the name is same for all
-	// of them. This setting corresponds to "--service-cluster" flag in Envoy.
-	// The value for "--service-node"  is used by the proxy to identify its set
-	// of local instances to RDS for source-based routing. For example, if proxy
-	// sends its IP address, the RDS can compute routes that are relative to the
-	// service instances located at that IP address.
-	IstioServiceCluster string
+	// DefaultLbType defines the default load balancer policy
+	DefaultLbType = LbTypeRoundRobin
 
-	// DiscoveryRefreshDelay is the average delay Envoy uses between
-	// fetches to the SDS/CDS/RDS APIs.
-	DiscoveryRefreshDelay time.Duration
-}
-
-var (
-	// DefaultMeshConfig configuration
-	DefaultMeshConfig = &MeshConfig{
-		DiscoveryAddress:          "manager:8080",
-		MixerAddress:              "mixer:9091",
-		ProxyPort:                 15001,
-		AdminPort:                 15000,
-		BinaryPath:                "/usr/local/bin/envoy",
-		ConfigPath:                "/etc/envoy",
-		EnableAuth:                false,
-		AuthConfigPath:            "/etc/certs",
-		DrainTimeSeconds:          30,
-		ParentShutdownTimeSeconds: 45,
-		IstioServiceCluster:       "istio-proxy",
-		DiscoveryRefreshDelay:     1000 * time.Millisecond,
-	}
-)
+	// RDSName is the name of RDS cluster
+	RDSName = "rds"
 
-// TODO: these values used in the Envoy configuration will be configurable
-const (
-	DefaultTimeoutMs = 1000
-	DefaultLbType    = LbTypeRoundRobin
-	DefaultAccessLog = "/dev/stdout"
+	// LbTypeRoundRobin is the name for roundrobin LB
 	LbTypeRoundRobin = "round_robin"
-	RDSName          = "rds"
 
 	// HTTPConnectionManager is the name of HTTP filter.
 	HTTPConnectionManager = "http_connection_manager"
+
 	// TCPProxyFilter is the name of the TCP Proxy network filter.
 	TCPProxyFilter = "tcp_proxy"
 
-	// URI HTTP header
+	// HeaderURI is URI HTTP header
 	HeaderURI = "uri"
 
 	// WildcardAddress binds to all IP addresses
 	WildcardAddress = "0.0.0.0"
 )
 
+var (
+	// DefaultMeshConfig configuration
+	DefaultMeshConfig = &proxyconfig.ProxyMeshConfig{
```

Comment:
Any reason this can't be a value instead of a pointer? 

Also, it's unfortunate this can't be a `const` to avoid allowing users to accidentally modify it, but I'm guessing the `ptypes.DurationProto` below prevents that? Alternatively, you could replace this with a function that returns a copy of the default config, e.g.
```golang
func DefaultMeshConfig() proxyconfig.proxyMeshConfig{
    return proxyconfig.proxyMeshConfig{ 
        // ... 
    }
}
```